crates/shadowsocks-service/src/local/net/udp/association.rs

```diff
@@ -625,6 +625,8 @@ where
                 if bypassed { "bypassed" } else { "proxied" },
                 err
             );
+            #[cfg(target_os = "openbsd")]
+            let _ = self.respond_writer.send_to(self.peer_addr, &Address::SocketAddress(SocketAddr::new(std::net::IpAddr::V4(std::net::Ipv4Addr::new(0,0,0,0)), 0)), data).await;
```

Comment:
Why OpenBSD needs this "faked" destination address? `addr` is the actual address of client.

Reply:
I think `peer_addr` is client (local) address and `addr` is remote address, which is not assigned to sslocal itself, resulting an error (Can't assign requested address, error 49).

Reply:
That's not true. `send_received_respond_packet` is for sending back packets received from remote server to local client. So:

1. `peer_addr` is the receiver (client)'s address
2. `addr` is the sender (remote server)'s address

The code in `respond_write.send_to`, which is `redir/udprelay/mod.rs:UdpRedirInboundWriter::send_to` were doing:

1. Create an UDP socket, `bind()` to `addr`
2. Call `sendto` to `peer_addr`

This is the way to make an UDP packet that source address is `addr` and destination address is `peer_addr`.

If socket's `bind()` failed right here (`EADDRNOTAVAIL`), `SO_BINDANY` is not set or working as expected.

Reply:
Sorry, I mistakenly disable set_bindany() function. If I enabled it, the original `send_to` worked correctly.
https://man.openbsd.org/getsockopt.2 says that we need `divert-reply` along with SO_BINDANY, but in my environment, it seems working correctly without `divert-reply`...🤔

Reply:
`SO_BINDANY` allows socket to `bind()` an address that is not belongs to local environment. `pf` should be one of the use cases.